### DISCUSSION TITLE
How to setup Budibase to be able to install community plugins and support questions

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#13084] [Help]
Hi all,

I am trying to understand from documentation how to install community plugins at self host but information is confusing. I cannot find a chapter that clearly explains how to install a community plugin. What configuration is needed at .env (PLUGINS_DIR=)? 
For example we want to test the [budibase-csv-import](https://github.com/mjashanks/budibase-csv-import) plugin but download from github link does not work and tar.gz is not available (maybe I am wrong). 
I downloaded zip from master. Extracted and then created a tar.gz with the contents of directory and it seams to be imported successfully, but I am not sure if this a the correct way.

Secondly I wonder what is the maintenance policy on community plugins by offecial Budibase side. I know they are community based and Budibase policy does not support custom plugins but I see they are created by the same developers working on Budibase. Can we trust community plugins to work at future Budibase updates or to use them means to be actively involved in their development?
Some of them could be so useful that should be part of official release. 

Thank you in advance for the clarifications.

**Comment by ConorWebb96** [comment]
Hey @gmentzik,

For your initial question, there's a helpful video explaining how to set up the CLI for custom [plugins](https://vimeo.com/754759041). The `PLUGINS_DIR=` variable just needs to be set to the relative path where your plugins will be located. (You don't need to do this to install plugins. You can copy and paste the GitHub URL into the plugins area to add the plugin to your self-hosted instance.)

![Screenshot](https://github.com/Budibase/budibase/assets/126772285/b760ca5f-8f00-491e-9c8f-270bc979e640)

The way you described is another method, and either way works. However, the alternative method I mentioned above is quicker, in my opinion.

Regarding your question about downloading, you can find the link to the `tar.gz` file within its releases [here](https://github.com/mjashanks/budibase-csv-import/releases). Below, I've also demonstrated this working directly from the URL.

![Demo](https://github.com/Budibase/budibase/assets/126772285/18b429e8-ffa7-4213-99fa-d4712ace4dc7)

As for your second question, these plugins are developed in our spare time, similar to other non-Budibase employees. Just like others, it's up to the developer to fix them. If they don't wish to, you can fork the repository, implement a fix, and then use that forked version going forward. I personally haven't seen any of the plugins break yet, but we can't guarantee that they will always work as we don't know what will happen in development.
